daemon/network/filter.go

```diff
@@ -46,7 +51,31 @@ func NewFilter(args filters.Args) (Filter, error) {
 	if err := args.WalkValues("type", validateNetworkTypeFilter); err != nil {
 		return Filter{}, err
 	}
-	return Filter{args: args, filterByUse: filterByUse, danglingOnly: danglingOnly}, nil
+	until := time.Time{}
+	if untilFilters := args.Get("until"); len(untilFilters) > 0 {
+		if len(untilFilters) > 1 {
+			return Filter{}, errdefs.InvalidParameter(errors.New("more than one until filter specified"))
+		}
+		ts, err := timestamp.GetTimestamp(untilFilters[0], time.Now())
+		if err != nil {
+			return Filter{}, errdefs.InvalidParameter(err)
+		}
+		seconds, nanoseconds, err := timestamp.ParseTimestamps(ts, 0)
```

Comment:
off-topic; perhaps we should rename this to `timestamp.Parse()` - I thought we did when moving, but maybe we left that for a follow-up exercise 😂